[DO NOT MERGE] DOCS-12770 lifecycle states

wwarner-salesforce opened the pull request:
DOCS-12770 lifecycle states

wwarner-salesforce committed:
DOCS-12770 lifecycle states

DOCS-12770 lifecycle states

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update nav.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update exchange-api.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update to-publish-assets-maven.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update to-search-with-graph-api.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

LucasLudueno commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,87 @@
+
+The Exchange user interface and APIs change some asset version information for asset versions in any state. This information is not changed by overwriting an asset version in the `development` state. This information includes:
+
+* tags
```
```suggestion
* asset type
* tags
```

wwarner-salesforce replied:
Thank you very much! Added "asset type" to the list.

LucasLudueno reviewed (commented):
only one comment. Same question for this one. I think we should not merge this PR this it will update the official documentation

wwarner-salesforce commented:
You're exactly right, Lucas! Adding that info now:

NOTE: This branch is part of the beta site: https://beta.docs.stgx.mulesoft.com/beta-lifecycle/exchange/lifecycle
At GA, currently planned for Feb 10, merge it into "latest" and it will be available on the public site: https://docs.mulesoft.com/exchange/ , specifically https://docs.mulesoft.com/exchange/lifecycle

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Update lifecycle.adoc

wwarner-salesforce committed:
Merge branch 'latest' into lifecycle

JennyHajee committed:
Update to-create-an-asset.adoc

added lifecycle states and new UI changes

JennyHajee committed:
update info for new UI changes

updated UI changes in to-create-an-asset.adoc and to-describe-an-asset.adoc

JennyHajee committed:
Update to-describe-an-asset.adoc

JennyHajee committed:
Update to-describe-an-asset.adoc

JennyHajee committed:
Update to-describe-an-asset.adoc

JennyHajee committed:
updated Simulate Data in REST APIs topic

 Simulate Data in REST APIs topic

JennyHajee committed:
Update to-download-an-asset.adoc

JennyHajee committed:
updated Manage Categories topic

updated Manage Categories topic

constanzaderienzo commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -113,6 +115,7 @@ To create an API asset:
 If the file is a ZIP, the selected main file must be in the root directory of the ZIP file. If the file is not a ZIP or if the file is a ZIP file with only one main file, then the main file is selected automatically.
 +
 . (Optional) Select *Advanced* and edit the advanced settings: *GroupId*, *AssetId*, *Version*, and *API version*. Exchange generates the group ID, asset ID, and version for you, and you can change these values here. You can change an API asset's version (asset version) and API version separately. The advanced settings are most often used to change the asset version.
+. Select an option for *Lifecycle state*.
```
I think this should be (Optional). By default, the Stable option is selected, so if nothing was clicked it would create it as a Stable asset.

constanzaderienzo commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -190,10 +194,19 @@ The name, icon, and description properties can only be changed in the Exchange a
-. In the sidebar *Versions* table, select *Add New Version*.
-. Enter the *Version* (asset version).
-. (Optional) Enter the *API version*.
+. CLick *Add version*. 
```
```suggestion
. Click *Add version*. 
```

constanzaderienzo commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -190,10 +194,19 @@ The name, icon, and description properties can only be changed in the Exchange a
-. Enter the *Version* (asset version).
-. (Optional) Enter the *API version*.
+. CLick *Add version*. 
+. For API assets, follow the steps for the corresponding API asst type:
```
```suggestion
. For API assets, follow the steps for the corresponding API asset type:
```

constanzaderienzo commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -190,10 +194,19 @@ The name, icon, and description properties can only be changed in the Exchange a
 If you have contributor or admin access to an asset, you can add a new version from the asset portal:
```
There is also another way to add a version through the versions table modal (manage versions button). It may be worth adding a note about that.

constanzaderienzo commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -12,90 +12,90 @@ Each minor version of an asset has its own tags, categories, custom fields, and
+When searching for assets, categories can be used to filter the search results. 
 
-NOTE: Before the August 2019 release, the versions of an asset could have different icons. After the August 2019 release, a new asset icon applies to all versions of the asset.
+. In the asset details pane, click *Add categories*. 
```
The "Add categories" is only shown when no categories are present. If a category was already applied it would read Categories (1) and the process to add a category would differ slightly. I'm not sure if we want to specify that as well.

constanzaderienzo commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -12,90 +12,90 @@ Each minor version of an asset has its own tags, categories, custom fields, and
 Tags can only be lower case, and upper case tags are converted to lowercase. Special characters are permitted. An asset can have any number of tags.
 
-. In the sidebar *Tags* list select *Add a tag*.
+. In the asset detail pane, click *Add tags*.
```
Same here

constanzaderienzo commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -12,90 +12,90 @@ Each minor version of an asset has its own tags, categories, custom fields, and
 
 
 === Add Secondary Pages
+YOu can add more pages to the asset portal.
```
```suggestion
You can add more pages to the asset portal.
```

sancheztavo commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+
+An asset in the `stable` state cannot add an asset in the `development` state as a dependency.
+
+If an asset in the `development` state depends on one or more other assets in the `development` state, Exchange does not prevent promoting that asset to the `stable` state. However, if a dependency in the `development` state is overwritten, the packages of the asset that was promoted to the `stable` state are not regenerated.
```
This text is not valid.

If an asset in the `development` state depends on one or more other assets in the `development` state, Exchange does not *allow* promoting that asset to the `stable` state, all its dependencies should first be promoted to `stable` first.

Apart from that, if you have an asset in `development` state with dependencies in `development` state and you modify a dependency, the dependent asset will also not be regenerated, it's the user's responsibility to republish the dependent asset once it's dependencies were updated

JennyHajee committed:
2nd part of changes

changes for:
nav.adoc
to-change-raml-version.adoc
to-manage-custom-fields.adoc

New topic:
manage-verions.adoc

JennyHajee committed:
UI updates

JennyHajee committed:
More new UI updated content

More new UI updated content

JennyHajee committed:
More new UI edits

More new UI edits

JennyHajee committed:
Addressed Gustavo's feedback

Updated lifecycle states dependencies promotion

JennyHajee committed:
Update modules/ROOT/pages/to-create-an-asset.adoc

will fix

Co-authored-by: Constanza De Rienzo <62706868+constanzaderienzo@users.noreply.github.com>

JennyHajee committed:
Update modules/ROOT/pages/to-describe-an-asset.adoc

Co-authored-by: Constanza De Rienzo <62706868+constanzaderienzo@users.noreply.github.com>

JennyHajee committed:
Edits per Coni's feedback

JennyHajee committed:
Merge branch 'lifecycle' of https://github.com/mulesoft/docs-exchange into lifecycle

LucasLudueno commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+
+Promote an asset version's state either in the asset's Exchange portal or by using the Exchange API.
+
+In the asset's portal, click the lifecycle state button that displays the current state, and select the new state of *Development*, *Stable*, or *Deprecated*.
```
```suggestion
In the asset's documentation page, click the lifecycle state button that displays the current state, and select the new state of *Development*, *Stable*, or *Deprecated*.
```

JennyHajee committed:
Update to-find-info.adoc

JennyHajee committed:
Added LCS to Manage Versions topic

JennyHajee committed:
Update to-find-info.adoc

JennyHajee committed:
Search for Assets updates

JennyHajee committed:
Update to-find-info.adoc

JennyHajee committed:
updated search graphic

JennyHajee committed:
restructured to-find-info.adoc

Restructured content for better flow and readability.

JennyHajee committed:
updated to-find-info.adoc

LucasLudueno commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,63 +5,39 @@ endif::[]
+
+Use Exchange to discover, share, and incorporate assets and resources into your applications. Exchange helps you create API developer portals, view and test APIs, simulate sending data to APIs by using the mocking service, create assets, and use API Notebooks to describe and test API functions.
+
+Exchange assets are accessible in Anypoint Studio (connectors, templates, and examples), API Manager (APIs, API groups, and policies), and Runtime Manager (APIs), and you can publish APIs from Design Center to Exchange.
```
```suggestion
Exchange assets are accessible in Anypoint Studio (connectors, templates, and examples), API Manager (APIs, API groups, and policies), and Runtime Manager (Applications), and you can publish APIs from Design Center to Exchange.
```

LucasLudueno commented on modules/ROOT/pages/to-use-api-notebook.adoc:
```diff
@@ -8,13 +8,13 @@ API Notebook lets you document an API's functionality with explanations that acc
-Note: An OAS specification is converted into RAML so you can follow the RAML conventions in API Notebook.
+Use API Designer in Anypoint Platform to improve and test your RAML, OAS, or AsyncAPI API specification, and to publish the API specification to Anypoint Exchange.
+
+NOTE: An OAS specification is converted into RAML so you can follow the RAML conventions in API Notebook.
```
and viceversa

JennyHajee committed:
Search for Assets Added back the original search experience

Added back the original search experience

JennyHajee committed:
Update to-find-info.adoc

fixed Search by Categories heading

JennyHajee committed:
small updates to asset details and describe an asset

JennyHajee committed:
Update to-find-info.adoc

JennyHajee committed:
updates to asset details, nav, landing page

JennyHajee committed:
Update asset-details.adoc

updated note

JennyHajee committed:
Update asset-details.adoc

JennyHajee committed:
updates per Ria

JennyHajee committed:
Update asset-details.adoc

fixes to asset details

JennyHajee committed:
Update asset-details.adoc

JennyHajee committed:
removed Deprecate Asset option from about-administration-tasks.adoc since manage versions replaces this.

JennyHajee committed:
Update exchange2-faq.adoc

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -4,9 +4,17 @@ include::_attributes.adoc[]
 :imagesdir: ../assets/images/
 
-When you click an asset, Anypoint Exchange provides detailed information.
+NOTE: The current documentation provides information about the new UI for the asset details page, which includes the lifecycle states feature. The previous UI does not include lifecycle states. 
```
Consider...
```suggestion
NOTE: Current product documentation includes information about the new lifecycle states feature visible in the asset details page of the updated UI.
```
HOWEVER...I'm not sure why this note is necessary; in fact, our style is to avoid beginning any topic with a notice. Could you incorporate this into the introductory text below? I think it goes without saying that this doc is updated to reflect changes in the UI....

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -4,9 +4,17 @@ include::_attributes.adoc[]
+NOTE: The current documentation provides information about the new UI for the asset details page, which includes the lifecycle states feature. The previous UI does not include lifecycle states. 
 
-The left navigation area lists:
+The asset details page provides comprehensive information about an asset. You can view version agnostic information, share and download, view code, manage versions and lifecycle states, add categories and tags, edit the documentation, and test API endpoints. 
```
```suggestion
The asset details page provides comprehensive information about an asset, including lifecycle states. You can view version-agnostic information, share and download, view code, manage versions and lifecycle states, add categories and tags, edit the documentation, and test API endpoints. 
```
Also, "share and download <what?>"

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -4,9 +4,17 @@ include::_attributes.adoc[]
-The left navigation area lists:
+The asset details page provides comprehensive information about an asset. You can view version agnostic information, share and download, view code, manage versions and lifecycle states, add categories and tags, edit the documentation, and test API endpoints. 
+
+See xref:workflow-sharing-assets.adoc[Workflow: Publish Assets] for more information about performing these functions.
```
Going forward, let's work to remove colons from topic titles.

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -15,31 +23,44 @@ The left navigation area lists:
 * The HTTP functions in the REST specification (for a REST API)
 * Terms and conditions and APIs using this policy (for a policy)
```
I know this is untouched but I suggest reordering the content of these list items as follows:
```suggestion
* For a REST API, the HTTP functions in the REST specification
* For a policy, the terms and conditions and APIs using this policy 
```

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -15,31 +23,44 @@ The left navigation area lists:
 * Terms and conditions and APIs using this policy (for a policy)
 
-The right detail panel lists:
+The top asset detail pane provides the following information:
```
Is there no label for "the top" pane?
(This is something I am hoping to change via work with our new Design System team....)

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -15,31 +23,44 @@ The left navigation area lists:
+* Asset creation date 
+* Last updated date
+* Version 
+* Categories  and tags of the asset
```
Remove space:
```suggestion
* Categories and tags of the asset
```

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -15,31 +23,44 @@ The left navigation area lists:
+* Version 
+* Categories  and tags of the asset
+* Lifecycle state
+* Share by inviting users to have access
+* For APIs, download specifications and view code
+* Add a new version
```
These list items are not parallel with their peers and do not logically complete the intro of things listed.

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -15,31 +23,44 @@ The left navigation area lists:
-* When the asset was last published
-* A list of asset patch versions
-* The tags and categories of the asset
+For REST APIs, use the right pane to test the API by creating example calls and test data (mocking), so you can send data to the API and view the results.
```
I'm concerned that the audience will be confused by "top pane," "right pane," "asset details pane," and so on. We need to make sure that each is distinct from the other and easily recognizable as what we are documenting in the UI.

kbostrom replied:
Also, what is the user goal in this sentence? Testing and viewing the results? There are so many actions going on (test API, create calls, test data, send data, view results) that the focus seems buried. How about...
```suggestion
For REST APIs, use the right pane to create example calls and mock data to test the API.
```
Can we call the right pane the testing environment? 
And the tester sends the results to...the API? Why would results be separate from he testing ground?

kbostrom replied:
I'm going to focus on only copyediting the rest of the changed content in this PR, because I don't think there is time to focus on context this time around.

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -15,31 +23,44 @@ The left navigation area lists:
+For REST APIs, use the right pane to test the API by creating example calls and test data (mocking), so you can send data to the API and view the results.
 
-For REST APIs, use the right panel to test the API by creating example calls and test data (mocking), so you can send data to the API and view the results.
+From the asset details pane, you can collapse and expand metadata such as major and minor versions, last published, and categories and tags. Use the *Manage versions* button to view a table of the asset patch versions. The table lists the version, instances, and the lifecycle state of dependent assets. You can add a new version of the asset, view version details, and delete a version. See xref:ma ...
```
This para has a lot of list items and it's kind of hard to track. Consider leading off with "version information" and then consolidate what you can see here and what you can do here (add or delete, in addition to viewing all this stuff).

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -15,31 +23,44 @@ The left navigation area lists:
+From the asset details pane, you can collapse and expand metadata such as major and minor versions, last published, and categories and tags. Use the *Manage versions* button to view a table of the asset patch versions. The table lists the version, instances, and the lifecycle state of dependent assets. You can add a new version of the asset, view version details, and delete a version. See xref:ma ...
 
-In the table of asset patch versions in the detail panel, each version has a drop-down menu with a *Version detail* option that shows information about that version, including who created it, when it was published, and any dependent assets.
+The documentation page displays the asset documentation. The documentation can be edited by an administrator or the asset owner by clicking *Edit documentation*. 
```
Is this in a separate location from the asset details page we have been describing?

kbostrom commented on modules/ROOT/pages/asset-details.adoc:
```diff
@@ -4,9 +4,17 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images/
 
```
After reading this text, I wonder why this topic title is Asset Details (indicating a reference topic) rather than something like Contents of the Asset Details Page (ref) or Viewing Asset Details (because there are "hidden" tasks in here about how to view, test, add, delete, and edit documentation)

kbostrom commented on modules/ROOT/pages/asset-types.adoc:
```diff
@@ -0,0 +1,29 @@
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images/
+
```
GLOBALLY replace space-hyphen-space with a colon.

kbostrom commented on modules/ROOT/pages/asset-types.adoc:
```diff
@@ -0,0 +1,29 @@
+
+The following asset types are supported in Exchange:
+
+* Connectors - Packaged connectivity to an endpoint developed and deployed on Anypoint Platform with third-party APIs and standard integration protocols
```
```suggestion
* Connectors: Packaged connectivity to an Anypoint Platform endpoint using third-party APIs and standard integration protocols
```

kbostrom commented on modules/ROOT/pages/asset-types.adoc:
```diff
@@ -0,0 +1,29 @@
+* Connectors - Packaged connectivity to an endpoint developed and deployed on Anypoint Platform with third-party APIs and standard integration protocols
++
+Use connectors within your application's flows to send and receive data using a protocol or specific API. Anypoint Studio comes with many bundled connectors, and Exchange has many more.
+* Templates - Packaged integration patterns built on best practices to address common use cases
```
```suggestion
* Templates: Packaged integration patterns that implement best practices to address common use cases
```

kbostrom commented on modules/ROOT/pages/asset-types.adoc:
```diff
@@ -0,0 +1,29 @@
+Use connectors within your application's flows to send and receive data using a protocol or specific API. Anypoint Studio comes with many bundled connectors, and Exchange has many more.
+* Templates - Packaged integration patterns built on best practices to address common use cases
++
+You can add your information such as user credentials to complete the template's use case or solution, and you can customize or extend templates as needed.
```
```suggestion
Complete the template's use case or solution by supplying your own information, customizing or extending the templates as needed.
```

kbostrom commented on modules/ROOT/pages/asset-types.adoc:
```diff
@@ -0,0 +1,29 @@
+* Policies - Configuration modules to extend the functionality of an API and enforce capabilities such as security
+* REST APIs - A RAML file or OAS file that specifies an API
++
+These APIs can be referenced by an HTTP Request connector to expose metadata to Anypoint Studio.
```
```suggestion
* REST APIs: RAML or OAS files that specify APIs referenced by an HTTP Request connector to expose metadata to Anypoint Studio
```

kbostrom commented on modules/ROOT/pages/asset-types.adoc:
```diff
@@ -0,0 +1,29 @@
+* API Groups - A set of APIs bundled into a single asset
++
+Instead of requesting access to multiple APIs to satisfy a use case, a developer can access the group in one step.
+* API Spec Fragments - A RAML document that has a version and an identifier, but is not in itself a complete RAML specification
```
```suggestion
* API Spec Fragments - A RAML document that has a version and an identifier but is not in itself a complete RAML specification
```

kbostrom commented on modules/ROOT/pages/asset-types.adoc:
```diff
@@ -0,0 +1,29 @@
+Instead of requesting access to multiple APIs to satisfy a use case, a developer can access the group in one step.
+* API Spec Fragments - A RAML document that has a version and an identifier, but is not in itself a complete RAML specification
++
+API spec fragments are also known as RAML fragments.
```
```suggestion
API spec fragments are also known as _RAML fragments_.
```

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+
+Find assets and refine search results in Exchange using the following filters that are available from the asset portal:
+
+*  Search all assets field
```
omit space
```suggestion
* Search all assets field
```

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+* Any organization
+* Any category
+* Any tag
+* Any Lifecycle state
```
```suggestion
* Any lifecycle state
```

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+
+== Search by Type
+
+Exchange supports the following asset types:
```
Is it necessary to repeat the list from "Asset Types"?
Also, note that the types are listed in a different order here than in the Asset Types topic.

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+
+To search for assets by type:
+
+. Click *Any type*.
```
Do we need to provide the user with context/navigation to the UI area in which the user clicks **Any type**?

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+To search for assets by type:
+
+. Click *Any type*.
+. Select from the list. You can select more than one type.
```
```suggestion
. Select one or more types from the list.
```

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+
+. Click *Any type*.
+. Select from the list. You can select more than one type.
+. Click *Apply*. The corresponding assets display in the search results.
```
```suggestion
. Click *Apply*. 
+
The corresponding assets display in the search results.
```

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+
+Find asset versions that are ready for consumption, in development, or have been deprecated.
+
+. From the details asset pane, click *Any Lifecycle state*.
```
This is what I mean by navigation context (see comment at Line 37 above).

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+. From the details asset pane, click *Any Lifecycle state*.
+. Select an option from the list and click *Apply*.
+. Assets in that state display in the search results.
+. (Optional) Click *Save this search* to create a shortcut for this search. Saved searches display in the left navigation bar.
```
GLOBALLY rephrase optional steps by beginning with the goal: an "if" statement:
```suggestion
. If you want to create a shortcut for this search, click *Save this search*. Saved searches display in the left navigation bar.
```

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+Find asset versions that are ready for consumption, in development, or have been deprecated.
+
+. From the details asset pane, click *Any Lifecycle state*.
+. Select an option from the list and click *Apply*.
+. Assets in that state display in the search results.
```
Reformat so that Line 168 is not a step.

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
++
+image::ex2-saved-searches-groups.png[Saved Search Groups]
+
+=== Rename or Delete a Saved Search
```
In general, you want to have more than one subtopic before creating subtopics. 
The alternatives are to include a solitary subtopic within its parent, or to create more than one: in this case, you could  have separate rename and delete subtopics?

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+
+=== Rename or Delete a Saved Search
+
+Saved searches can be renamed or deleted.
```
As a followup to the previous comment, and in general, consider telling the reader under what circumstances they might want or need to perform this task, such as...:

"If <this happens>, rename your search."
"To save system resources, delete your search when you no longer need it."

kbostrom commented on modules/ROOT/pages/do-not-use-to-find-info-with-new-search.adoc:
```diff
@@ -0,0 +1,205 @@
+
+To select the ranking of search results, use the *Sort by* drop-down menu to choose *Relevance* (default), *Name*, *Last modified*, or *Rating*. 
+
+Relevance sorts the matching results by where a match is found in the asset metadata, such as name, title, pages, and API specification metadata. API specification metadata includes endpoints and types.
```
***BREAK edit here to commit and continue***

kbostrom reviewed (dismissed):
First portion of edit review complete.

kbostrom commented on modules/ROOT/pages/ex2-to-simulate-api-data.adoc:
```diff
@@ -6,17 +6,18 @@ endif::[]
 Exchange lets you provide simulated data for testing OAS or RAML REST APIs. The simulated data is provided in the Mocking Service feature.
 
-The Mocking Service appears in Exchange:
+The following example shows how the Mocking Service simulates the data for a REST API:
```
Per our SSoT Product Tiering and Hierarchy reference (https://docs.google.com/spreadsheets/d/1f6v9cjx84ZAHGeZs36A0clZoQc_ggAe6m6e-NQ9BUuk/edit#gid=71218867) and Term Presentation List (https://confluence.internal.salesforce.com/display/MTDT/Term+Presentation+Reference#TermPresentationReference-M), lowercase "mocking service," because it is not a MuleSoft-specific name.
```suggestion
The following example shows how the mocking service simulates the data for a REST API:
```

kbostrom commented on modules/ROOT/pages/ex2-to-simulate-api-data.adoc:
```diff
@@ -6,17 +6,18 @@ endif::[]
 * In the API console that appears on the right side of an asset detail screen. The Mocking Service lets you add simulated data to an API function and test the API with the data.
 +
-image::ex2-api-console.png[Mocking in API Console]
+image::exchange-rest-api-mocking-service.png[Mocking  in API Console]
```
remove space
```suggestion
image::exchange-rest-api-mocking-service.png[Mocking in API Console]
```

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -239,6 +243,37 @@ curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v2/or
+[[set-lifecycle-state]]
+=== Set Lifecycle State
+
+A new asset version is in the `development` state or the default `stable` state. The `stable` state is also known as `published`. An asset version must be in the `stable` state before it is in the `deprecated` state.
```
```suggestion
A new asset version is in either the `development` state or the default `stable` state (also known as `published`). An asset version must be stable before being deprecated.
```

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -239,6 +243,37 @@ curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v2/or
+
+A new asset version is in the `development` state or the default `stable` state. The `stable` state is also known as `published`. An asset version must be in the `stable` state before it is in the `deprecated` state.
+
+To use the Exchange API version 2 to create an asset and set its lifecycle state, add the property `status` to the multipart request. If the state is not specified, the asset is created with the default `stable` state.
```
```suggestion
To use Exchange API version 2 to create an asset and set its lifecycle state, add the property `status` to the multipart request. If the state is not specified, the asset is created with the default `stable` state.
```
...or do we mean "version 2" to indicate different implementations of the API rather than an official version number? Whether to use "the" depends on the answer.

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -239,6 +243,37 @@ curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v2/or
+
+To use the Exchange API version 2 to create an asset and set its lifecycle state, add the property `status` to the multipart request. If the state is not specified, the asset is created with the default `stable` state.
+
+REST API asset example:
```
Reword to be a complete sentence intro, something like "The following example shows <how to?><a typical?>...."

kbostrom replied:
Also, after seeing the subsequent text about version 1, consider adding version 2 to this intro sentence.
I'm a bit confused about versions 1 and 2: are these product versions or instance example numbers?

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -239,6 +243,37 @@ curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v2/or
+
+. Log in to Anypoint Platform and download the https://anypoint.mulesoft.com/exchange/organizations/68ef9520-24e9-4cf2-b2f5-620025690913/assets/68ef9520-24e9-4cf2-b2f5-620025690913/training-american-flights-api/1.0.0/files/fat-raml/zip[example file].
+. Replace `/file-path/training-american-flights-api-1.0.0-raml.zip` with the local path of the downloaded file.
+. Replace `ANYPOINT_TOKEN`, `:organizationId`, `:groupId`, `:assetId`, and `:version` with your information.
```
```suggestion
. Replace `ANYPOINT_TOKEN`, `:organizationId`, `:groupId`, `:assetId`, and `:version` with values specific to your implementation.
```

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -239,6 +243,37 @@ curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v2/or
+. Log in to Anypoint Platform and download the https://anypoint.mulesoft.com/exchange/organizations/68ef9520-24e9-4cf2-b2f5-620025690913/assets/68ef9520-24e9-4cf2-b2f5-620025690913/training-american-flights-api/1.0.0/files/fat-raml/zip[example file].
+. Replace `/file-path/training-american-flights-api-1.0.0-raml.zip` with the local path of the downloaded file.
+. Replace `ANYPOINT_TOKEN`, `:organizationId`, `:groupId`, `:assetId`, and `:version` with your information.
+. Set the property `status` to `development` or `published`.
```
Yikes, do we really have to switch between "stable" and "published"? If we must introduce both terms, after doing so, we should pick one and use it from the introduction onward.

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -239,6 +243,37 @@ curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v2/or
+    -F 'files.raml.zip=@/file-path/training-american-flights-api-1.0.0-raml.zip' \   https://anypoint.mulesoft.com/exchange/api/v2/organizations/:organizationId/assets/:groupId/:assetId/:version
+----
+
+To use the Exchange API version 1 to create an asset and set its lifecycle state, send the status property as a field.
```
Why does this mention creation as well as lifecycle state?
Do we need separate full examples for v1 and v2, or does this substitution make sense to your audience?

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -295,6 +330,26 @@ https://anypoint.mulesoft.com/exchange/api/v2/assets/:groupId/:assetId/:version/
+. `stable` to `deprecated`
+. `deprecated` to `stable`
+
+The `stable` state is also known as `published`.
```
See previous comment re. synonyms/equivalent wordings: I suggest introducing both but subsequently using only one in the narrative, if possible.

kbostrom commented on modules/ROOT/pages/exchange-api.adoc:
```diff
@@ -295,6 +330,26 @@ https://anypoint.mulesoft.com/exchange/api/v2/assets/:groupId/:assetId/:version/
+
+The `stable` state is also known as `published`.
+
+In this example, replace `ANYPOINT_TOKEN`, `:groupId`, `:assetId`, and `:version` with your information, and replace `:status` with `development`, `published`, or `deprecated`.
```
```suggestion
In this example, replace `ANYPOINT_TOKEN`, `:groupId`, `:assetId`, and `:version` with values specific to your implementation, and replace `:status` with one of `development`, `published`, or `deprecated`.
```

kbostrom commented on modules/ROOT/pages/exchange2-faq.adoc:
```diff
@@ -58,7 +58,7 @@ You can use RAML data types, but not a JSON referenced by another JSON schema.
 Exchange only works in Windows with the latest versions of Chrome, Firefox, and Internet Explorer 11. IE 9 is not supported and displays Exchange incorrectly.
 
-== When I push a custom connector through Maven to Exchange, how can i define a custom icon?
+== When I push a custom connector through Maven to Exchange, how can I define a custom icon?
```
See style guide entry regarding the use of FAQs; going forward, include this information in appropriate locations in the narrative so that the question does not arise in the first place.

IsaacEldridge replied:
If this is a heading, shouldn't it be camel case capitalization?

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
+
+Exchange assets are accessible in Anypoint Studio (connectors, templates, and examples), API Manager (APIs, API groups, and policies), and Runtime Manager (applications), and you can publish APIs from Design Center to Exchange.
+
+Assets under the *Provided by MuleSoft* organization are created by MuleSoft and its partners, and are certified by Mulesoft. These assets are publicly available to all customers as well. However, all assets that are created by an organization and published into the organization's instance of Exchange, are private and not visible outside that organization. Customers are responsible for any privat ...
```
```suggestion
Assets in the *Provided by MuleSoft* organization are created by MuleSoft and its partners and certified by Mulesoft. These assets are publicly available to all customers. However, assets that are created by an organization other than *Provided by MuleSoft* and published in that organization's instance of Exchange are private and not visible outside that organization. Customers are responsible for any private assets they publish within their Exchange instance.
```
Should we explain what it means to be "responsible for"?

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
+Assets under the *Provided by MuleSoft* organization are created by MuleSoft and its partners, and are certified by Mulesoft. These assets are publicly available to all customers as well. However, all assets that are created by an organization and published into the organization's instance of Exchange, are private and not visible outside that organization. Customers are responsible for any privat ...
 
 == Get Started
 
```
It is best to provide some narrative context between the title and Step 1 of a task. Be sure to add value to the context beyond restating the title.

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
-. From the public https://www.anypoint.mulesoft.com/exchange/[Exchange], click an asset and view its description.
-. Click Login to view a private Exchange for your organization.
-. After you log in, click Public Portal in your organization to view public APIs available to anyone.
+. From https://www.anypoint.mulesoft.com/exchange/[Exchange], search for assets. If you know the name of the asset, enter the name in the *Search* field. 
```
```suggestion
. From https://www.anypoint.mulesoft.com/exchange/[Exchange], either search for assets or enter the name of a known asset in the *Search* field. 
```

kbostrom replied:
It is not always possible, but it is best to limit steps in a task to one sentence.

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
-. Click Login to view a private Exchange for your organization.
-. After you log in, click Public Portal in your organization to view public APIs available to anyone.
+. From https://www.anypoint.mulesoft.com/exchange/[Exchange], search for assets. If you know the name of the asset, enter the name in the *Search* field. 
+. Select an asset from the search results view.
```
If you search for a known asset, is only that one asset shown? Does this step apply to both circumstances provided in Step 1?

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
+. Review the description of the asset.
 
-After you log into Exchange, you can create assets for APIs, connectors, templates, examples, API Notebooks, and custom assets.
+To create assets for APIs, connectors, templates, examples, API Notebooks, and custom assets, log into Exchange. See xref:workflow-sharing-assets.adoc[Workflow: Publish APIs] for more information.
```
Why are we including creation info in what has so far been a search task?
Maybe this Get Started <with what?> topic should begin by saying that you can get started either by creating a new asset for using an existing one, and then go on from there?

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
-After you log into Exchange, you can create assets for APIs, connectors, templates, examples, API Notebooks, and custom assets.
+To create assets for APIs, connectors, templates, examples, API Notebooks, and custom assets, log into Exchange. See xref:workflow-sharing-assets.adoc[Workflow: Publish APIs] for more information.
 
 == Try APIs
```
I don't know what this title means. "Test" might be a better word, or "Find the Right API for Your Environment"?
Also could use a transition from title to Step 1.

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
 == Try APIs
 
-. Click All Types > REST APIs and choose any API.
+. Click the *All Types* filter, check the REST APIs option, and click *Apply*.
```
```suggestion
. Click the *All Types* filter, select the REST APIs option, and click *Apply*.
```
Per the style ref, use "select" and "unselect" for check boxes.

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
+. Click the *All Types* filter, check the REST APIs option, and click *Apply*.
 . Click the asset and notice how the API functions appear in the left
-navigation area. Exchange lets you view the functions in an API.
+navigation bar. Exchange lets you view the functions in an API.
```
Does the second sentence add anything to the first?

kbostrom commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+
+To use the Exchange API, follow the xref:exchange-api.adoc#promote-lifecycle-state[Promote Lifecycle State] example.
+
+== Dependencies
```
Can this title be made more robust? What, specifically, about dependencies, or what type of dependencies?
I'd at least add "State" to the title.

kbostrom replied:
...and do we need some text explaining what dependencies are and why your audience should care?

kbostrom commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+
+== Dependencies
+
+An asset in the `stable` state cannot add an asset in the `development` state as a dependency.
+
+If an asset in the `development` state depends on one or more other assets in the `development` state, Exchange does not allow promoting that asset to the `stable` state. All dependencies of the asset should be promoted to stable first. If an asset in the `development` state has dependencies in `development` state and the dependent is modified, the dependent asset is also not regenerated. It is t ...
```
```suggestion
All dependencies of an asset must be promoted before that asset can be promoted. For example, an asset in the `development` state cannot be promoted to the `stable` state until its dependencies are stable.
+
```
The second half of this is confusing: what are "regeneration" and "republishing" and how do they relate to asset state promotion? What do we mean by "...and the dependent is modified"? Does that mean promoted? Are we saying that the promotion of a dependency requires the asset owner to regenerate/republish the dependency for the promotion to be committed/official?

sancheztavo replied:
What must be clear is that you might have an asset that has embedded `development` dependencies in its files. If one of those dependencies is modified, the dependent asset will still have a previous version of the dependencies and needs to be republished in order for those files to be updated.

An example would be RAML APIs using RAML fragments published through API Designer

kbostrom commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+
+To create an asset using the Exchange Maven Facade API, follow the xref:to-publish-assets-maven.adoc#asset-lifecycle-state[Asset Lifecycle State] example.
+
+== Search Using the Graph API by Lifecycle State
```
The narrative mentions nothing about a graph; reword for clarity, or add text about a graph named "API by Lifecycle"?
Do we mean that in addition to finding an asset by search/selecting its name, we can also find it by its state? If so, I think that his text might belong in the topic about searching for assets.

kbostrom commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+* xref:exchange-api.adoc[Publish Assets Using the Exchange API]
+* xref:to-publish-assets-maven.adoc[Publish Assets Using Maven]
+* xref:to-search-with-graph-api.adoc[Search Using the Graph API]
+* https://anypoint.mulesoft.com/graph/api/v2/graphql[Graph API Web Interface]
```
**BREAK to commit**

kbostrom replied:
Retry commit; concurrent reviews causing connection issues.

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
+navigation bar. Exchange lets you view the functions in an API.
 . Click a function and more information appears. At the right side of the screen, you can test the API by entering sample values to send to the API and get responses from the API.
 
 == Need More Information on an Asset?
```
Our style is to not use sentences, especially questions, as titles.

kbostrom commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,48 @@ endif::[]
 
 == Versions
 
 This page applies to the current version, Exchange 2.
```
Please reconcile this with "the API Exchange version 2" (or 1) in other parts of the narrative

kbostrom reviewed (commented):
Second part of review

LucasLudueno commented on modules/ROOT/pages/to-publish-assets-maven.adoc:
```diff
@@ -279,6 +279,30 @@ Create an `assembly.xml` file in `src/main/resources/docs-assembly`:
+. Open your Exchange home page and view the template in the `development` state.
+
+If you execute `mvn clean deploy` multiple times, the asset is overwritten without any errors.
+
```
```suggestion

Note: an asset that is in development state and is not an SNAPSHOT version can not be consumed through Maven
```

@JennyHajee can we add a clarification like this?

JennyHajee committed:
Update modules/ROOT/pages/lifecycle.adoc

Co-authored-by: Lucas Ludueño <ludueno.lucas@gmail.com>

JennyHajee committed:
Update modules/ROOT/pages/ex2-to-simulate-api-data.adoc

Co-authored-by: Kim Bostrom <18537445+kbostrom@users.noreply.github.com>

JennyHajee committed:
Updated with Lucas's feedback

Added note

JennyHajee committed:
Merge branch 'latest' into lifecycle

JennyHajee committed:
updates for Kim's review

JennyHajee committed:
Merge branch 'lifecycle' of https://github.com/mulesoft/docs-exchange into lifecycle

JennyHajee committed:
Update asset-details.adoc

çhanged owner name and email to contact owner and email

JennyHajee committed:
Update asset-details.adoc

reworked line 45

constanzaderienzo commented on modules/ROOT/pages/to-find-info.adoc:
```diff
@@ -223,12 +231,10 @@ the reason is provided.
 |Optimized API |optimized-api |sales:/connect |Yes
 |===
 
+== Search by Types, Tags, Categories, and Custom Fields in the Orginal  Exchange Search Experience
```
```suggestion
== Search by Types, Tags, Categories, and Custom Fields in the Orginal Exchange Search Experience
```

JennyHajee committed:
Update lifecycle.adoc

constanzaderienzo commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,63 @@
+
+. Click *Manage versions* in the expanded asset details pane.
+. Click the three dots menu and choose *Version detail* from the menu.
+. Click *Close*.
```
Should this step be included? And if so, the first button to go back is called *Back* and then *Close* to close the modal.

constanzaderienzo commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,63 @@
+You can delete versions of an asset. If you try to delete an asset version with dependent assets, Exchange shows a warning. Remove the dependencies and then delete the asset version.
+
+. Click *Manage versions* in the expanded asset details pane.
+. Click the three dots menu and choose *Delete version* from the menu.
```
A confirmation modal is shown.
![image](https://user-images.githubusercontent.com/62706868/156625326-c5700c71-f1db-48a9-a706-d5d62af61141.png)

constanzaderienzo commented on modules/ROOT/pages/publish-an-http-api.adoc:
```diff
@@ -5,14 +5,15 @@ endif::[]
-. In the asset view, click *Edit* to provide a description.
-. Click *Save As Draft* and *Publish*.
+. In the asset view, click *Edit documentation* to provide a description.
+. Click *Save* and *Publish*.
```
Clicking Save is not needed if the final goal is to publish

constanzaderienzo commented on modules/ROOT/pages/publish-an-http-api.adoc:
```diff
@@ -5,14 +5,15 @@ endif::[]
 . Click *Publish*.
-. In the asset view, click *Edit* to provide a description.
-. Click *Save As Draft* and *Publish*.
+. In the asset view, click *Edit documentation* to provide a description.
```
To edit the description of an asset what they should click is the pencil icon

sancheztavo commented on modules/ROOT/pages/index.adoc:
```diff
@@ -5,75 +5,46 @@ endif::[]
+
+Use Exchange to discover, share, and incorporate assets and resources into your applications. Exchange helps you create API developer portals, view and test APIs, simulate sending data to APIs by using the mocking service, create assets, and use API Notebooks to describe and test API functions.
+
+Exchange assets are accessible in Anypoint Studio (connectors, templates, and examples), API Manager (APIs, API groups, and policies), and Runtime Manager (applications), and you can publish APIs from Design Center to Exchange.
```
I wouldn't enumerate here since it's not an exhaustive list, assets are also accessible in other ways such as Graph Service, Maven Facade, Design Center API Designer

LucasLudueno replied:
```suggestion
Exchange assets are accessible from many places like Anypoint Studio (connectors, templates, and examples), API Manager (APIs, API groups, and policies), or Runtime Manager (applications). APIs can be published from Design Center to Exchange.
```
@sancheztavo what so you think?

sancheztavo commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+
+Overwriting an asset version also updates the search index.
+
+The Exchange user interface and APIs change some asset version information for asset versions in any state. This information is not changed by overwriting an asset version in the `development` state. This information includes:
```
This paragraph doesn't make sense to me, I'd just add that there is information that is not overwritten by republishing an asset in development state, and list it

sancheztavo commented on modules/ROOT/pages/to-publish-assets-maven.adoc:
```diff
@@ -279,6 +279,32 @@ Create an `assembly.xml` file in `src/main/resources/docs-assembly`:
+
+A `SNAPSHOT` asset is permanently in the `development` state and cannot be promoted to any other state.
+
+An asset with the version `SNAPSHOT` published with the Anypoint Exchange Maven Facade API version 1 or version 2 has the state `stable`.
```
Maybe we can add that this behavior is aimed to be backwards compatible with the legacy APIs

JennyHajee committed:
Update modules/ROOT/pages/to-find-info.adoc

removing extra space

Co-authored-by: Constanza De Rienzo <62706868+constanzaderienzo@users.noreply.github.com>

LucasLudueno commented on modules/ROOT/pages/lifecycle.adoc:
```diff
@@ -0,0 +1,94 @@
+
+The public portal does not show asset versions in the `development` state.
+
+Anypoint Platform products other than Exchange, such as Anypoint API Designer, Anypoint Flow Designer, or Anypoint Studio, do not consume asset versions in the `development` state.
```
```suggestion
Anypoint Platform products other than Exchange, such as Anypoint Flow Designer, or Anypoint Studio, do not consume asset versions in the `development` state.
```

JennyHajee committed:
updates per Coni

JennyHajee committed:
Merge branch 'lifecycle' of https://github.com/mulesoft/docs-exchange into lifecycle

JennyHajee committed:
Gustavo and Lucas last changes

kbostrom commented on modules/ROOT/pages/to-use-api-notebook.adoc:
```diff
@@ -8,13 +8,13 @@ API Notebook lets you document an API's functionality with explanations that acc
-Note: An OAS specification is converted into RAML so you can follow the RAML conventions in API Notebook.
+Use API Designer in Anypoint Platform to improve and test your RAML, OAS, or AsyncAPI API specification, and to publish the API specification to Anypoint Exchange.
+
+NOTE: An OAS specification is converted into RAML so you can follow the RAML conventions in API Notebook. Also, the RAMl specification is converted when you upload an OAS.
```
Do you mean this?
```suggestion
NOTE: An OAS specification is converted into RAML so you can follow the RAML conventions in API Notebook. Also, the RAMl specification is converted when you upload an OAS specification.
```

kbostrom commented on modules/ROOT/pages/to-share-api-asset-to-portal.adoc:
```diff
@@ -25,11 +25,9 @@ Follow these steps to add an asset to a portal and make the asset public.
-. Select *Share* -> *Add to the public portal*.
-. (Optional) If you decide not to put the asset in the public portal, select *Share Internally* to return to the previous *Share* menu.
-. (Optional) If you wish to manage instance visibility before proceeding, select *API Instances*. After following this link, you are returned to the previous *Share* menu.
+. Select *Share* and choose *Public Publish it on the public portal*.
```
Is that verbatim of the UI: "Public Publish it on the public portal"?

kbostrom commented on modules/ROOT/pages/to-publish-assets-maven.adoc:
```diff
@@ -279,6 +279,32 @@ Create an `assembly.xml` file in `src/main/resources/docs-assembly`:
+
+If you execute `mvn clean deploy` multiple times, the asset is overwritten without any errors.
+
+Note: An asset that is in development state and is not an SNAPSHOT version cannot be consumed through Maven.
```
a SNAPSHOT version
an ESS EN A PEE ESS ACH OH TEE version

kbostrom commented on modules/ROOT/pages/to-publish-assets-maven.adoc:
```diff
@@ -279,6 +279,32 @@ Create an `assembly.xml` file in `src/main/resources/docs-assembly`:
+
+In a terminal:
+
+. Execute: `git clone https://github.com/mulesoft-labs/exchange-documentation-samples.git`
```
Why the colon after "Execute"?

kbostrom commented on modules/ROOT/pages/to-find-info.adoc:
```diff
@@ -141,10 +142,17 @@ This API is found by these searches:
+
+Using the original Exchange search experience, to search for case-insensitive text that begins with each word you specify, put a space between words. This is similar to putting the `*` wildcard regular expression after each word. For example, searching for `mq module` finds any asset that has text that starts with `mq` or `module`.
+
+Searching for a multiple word phrase shows assets that have every word in the phrase.
```
```suggestion
Searching for a multiple-word phrase shows assets that have every word in the phrase.
```

kbostrom commented on modules/ROOT/pages/to-find-info.adoc:
```diff
@@ -6,33 +6,34 @@ endif::[]
+To filter search results by type, categories, tags,and lifecycle states in the new Exchange search experience, use the drop-down menus on the search page.
 
-To select the ranking of search results in the new Exchange search experience, use the *Sort by* drop-down menu to choose *Relevance* (default), *Name*, *Last modified*, or *Rating*. *Relevance* sorts the matching results by where a match is found in the asset metadata, such as name, title, pages, and API specification metadata. API specification metadata includes endpoints and types.
+To select the ranking of search results in the new Exchange search experience, use the Sort by drop-down menu to choose Relevance (default), Name, Last modified, or Rating. Relevance sorts the matching results by where a match is found in the asset metadata, such as name, title, pages, and API specification metadata. API specification metadata includes endpoints and types.
```
The original presentation values are correct; revert.

kbostrom commented on modules/ROOT/pages/to-find-info.adoc:
```diff
@@ -6,33 +6,34 @@ endif::[]
+image:exchange-new-search.png[Exchange search experience]
 
-== Tags, Categories, and Ranking in the New Exchange Search Experience
+== Types, Categories, Tags, Lifecycle States and Ranking in the New Exchange Search Experience
```
```suggestion
== Types, Categories, Tags, Lifecycle States, and Ranking in the New Exchange Search Experience
```

kbostrom commented on modules/ROOT/pages/to-find-info.adoc:
```diff
@@ -6,33 +6,34 @@ endif::[]
+. Each asset’s name
+. Tags
+. Categories
+. Custom attributes (also known as custom fields)
```
```suggestion
. Custom attributes (also known as _custom fields_)
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
+. Select *Edit documentation*.
+. In the left navigation bar, click *Add New Page*. 
+. Press the enter or return key on your keyboard to create the page. You can create as many pages as needed.  
+.  To delete a page, click *Delete Page*.
```
```suggestion
. To delete a page, click *Delete Page*.
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
 Alternatively, select the image icon in the editor and provide a URL to an illustration available on the web.
 
-In Markdown, use this syntax to reference a web served image:
+In the *Markdown* tab, use this syntax to reference a web served image:
```
```suggestion
In the *Markdown* tab, use this syntax to reference a web-served image:
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
-. Select *Edit*.
-. Select *Visual mode*.
+. Click *Edit documentation*.
+. Select *Visual* tab.
```
```suggestion
. Select the *Visual* tab.
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
+. In the editor, you can change the appearance of text and insert code samples, block quotes, and horizontal lines.
 . Add images and videos as described below.
-. After you write the main page, select either *Discard changes* or *Save as draft*. You can edit and save a draft any number of times before publishing. Drafts are not visible to other users until they are published. When an asset has an unpublished draft, Exchange displays a *Go to draft* button to edit it.
+. After you write the main page, click *Save* to save as a draft or *Publish*. You can edit and save a draft any number of times before publishing. Drafts are not visible to other users until they are published. 
```
```suggestion
. After you write the main page, click *Save* to save as a draft or *Publish* to publish. You can edit and save a draft any number of times before publishing. Drafts are not visible to other users until they are published. 
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
+. In the asset details pane, click *Add tags*. If there are already tags added for the asset, click *Tags*.
+. Click *Add a tag* and select from the list of tags. 
 . Press the enter or return key on your keyboard.
+. To remove tags, click *Tags*, then click the *X* next to the tag text.
```
```suggestion
. To remove tags, click *Tags*, and then click the *X* next to the tag text.
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
+== Add and Remove Asset Tags
 
-== Edit Asset Description
+When searching for assets, tags can be used to filter the search results. Tags can only be lower case, and upper case tags are converted to lowercase. Special characters are permitted. 
```
"you" are searching, not "assets"
```suggestion
When searching for assets, you can use tags to filter the search results. Tags can use only lowercase letters, and uppercase tags are converted to lowercase. Special characters are permitted. 
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
-. Select the pen icon beside the asset icon.
-. Select an image.
-. (Optional) To remove an asset icon, select the cancel icon beside the asset icon.
+When searching for assets, categories can be used to filter the search results. 
```
"you" are searching, not "categories":
```suggestion
When searching for assets, you can use categories to filter the search results. 
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
+. To cancel the changes, click the *X* icon.
+. To save the changes, click the *check* icon.
 
+NOTE: Before the August 2019 release, the versions of an asset could have different names or icons. After the August 2019 release, a updated asset information applies to all versions of the asset.
```
```suggestion
NOTE: Before the August 2019 release, the versions of an asset could have different names or icons. After the August 2019 release, updated asset information applies to all versions of the asset.
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
+. To change the description, enter a new asset description in the *Description* field.
+. To change the contact name or email, enter a new contact name and email in the corresponding field.
+. To cancel the changes, click the *X* icon.
+. To save the changes, click the *check* icon.
```
Does the icon actually show the word "check" or do you mean a check symbol?

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
-NOTE: Before the August 2019 release, the versions of an asset could have different names. After the August 2019 release, a new name applies to all versions of the asset.
+. Click the edit pen icon in the asset portal.
+. To change the icon, click the edit pen icon, select an image file, then click *Open*.
+. To change the asset portal name, enter the new asset portal name.
```
```suggestion
. To change the asset portal name, enter a new asset portal name.
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
 
-NOTE: Before the August 2019 release, the versions of an asset could have different names. After the August 2019 release, a new name applies to all versions of the asset.
+. Click the edit pen icon in the asset portal.
+. To change the icon, click the edit pen icon, select an image file, then click *Open*.
```
```suggestion
. To change the icon, click the edit pen icon, select an image file, and then click *Open*.
```

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
+You can change the asset icon, name, description, contact name, and contact email. 
 
-NOTE: Before the August 2019 release, the versions of an asset could have different names. After the August 2019 release, a new name applies to all versions of the asset.
+. Click the edit pen icon in the asset portal.
```
We should try to show this icon rather than describe it in terms that might have different implications for different audiences.

kbostrom commented on modules/ROOT/pages/to-describe-an-asset.adoc:
```diff
@@ -4,98 +4,94 @@ include::_attributes.adoc[]
 :imagesdir: ../assets/images/
 
-You can edit an asset's portal; change its name, icon, description, tags, categories, and custom fields; and add portal pages with content.
+You can edit an asset's portal, change its name, icon, description, tags, categories, and custom fields; and add portal pages with content.
```
STET original: you must separate list items that have internal commas by using semicolons:
```suggestion
You can edit an asset's portal;
change its name, icon, description, tags, categories, and custom fields; 
and add portal pages with content.
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
+* Select the main file of the API.
 . (Optional) Choose a file to upload.
+. Enter the *Version* (asset version).
+. (Optional) Select on option for *Lifecycle state*. The default is *Stable*.
```
```suggestion
. Optionally select an option for *Lifecycle state*. The default is *Stable*.
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
+* Select Choose File to locate the API specification file.
+* Select the main file of the API.
 . (Optional) Choose a file to upload.
+. Enter the *Version* (asset version).
```
```suggestion
. Enter a *Version* value (asset version).
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
+. For RAML, API Spec Fragment, OAS, and WSDL assets:
+* Select Choose File to locate the API specification file.
+* Select the main file of the API.
 . (Optional) Choose a file to upload.
```
```suggestion
. Optionally, choose a file to upload.
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
+* API Spec Fragment - RAML : Provide an API Fragment RAML specification file. Fragment specifications must be a RAML file (.raml).
+* HTTP API - This asset does not require a file. This type of asset provides an API endpoint that is defined by API Manager.
+. For RAML, API Spec Fragment, OAS, and WSDL assets:
+* Select Choose File to locate the API specification file.
```
```suggestion
* Select *Choose File* to locate the API specification file.
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
+* REST API - RAML: Provide a RAML API specification file. * RAML specifications must be a RAML file (.raml).
+* REST API - OAS: Provide an OAS API specification file. *OAS specifications can be either a YAML (.yaml) or JSON (.json) file. Exchange supports OAS 2.0 and OAS 3.0 specifications.
+* SOAP API - WSDL: Provide a WSDL API specification file. SOAP specifications file can be either a WSDL (.wsdl) or XML (.xml) file.
+* API Spec Fragment - RAML : Provide an API Fragment RAML specification file. Fragment specifications must be a RAML file (.raml).
```
```suggestion
* API Spec Fragment - RAML: Provide an API Fragment RAML specification file. A fragment specification must be a RAML file (.raml).
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
+. For API assets, follow the steps for the corresponding API asset type:
+* REST API - RAML: Provide a RAML API specification file. * RAML specifications must be a RAML file (.raml).
+* REST API - OAS: Provide an OAS API specification file. *OAS specifications can be either a YAML (.yaml) or JSON (.json) file. Exchange supports OAS 2.0 and OAS 3.0 specifications.
+* SOAP API - WSDL: Provide a WSDL API specification file. SOAP specifications file can be either a WSDL (.wsdl) or XML (.xml) file.
```
```suggestion
* SOAP API - WSDL: Provide a WSDL API specification file. A SOAP specifications file can be either a WSDL (.wsdl) or XML (.xml) file.
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
+. Click *Add version*. 
+. For API assets, follow the steps for the corresponding API asset type:
+* REST API - RAML: Provide a RAML API specification file. * RAML specifications must be a RAML file (.raml).
+* REST API - OAS: Provide an OAS API specification file. *OAS specifications can be either a YAML (.yaml) or JSON (.json) file. Exchange supports OAS 2.0 and OAS 3.0 specifications.
```
```suggestion
* REST API - OAS: Provide an OAS API specification file. An OAS specification can be either a YAML (.yaml) or JSON (.json) file. Exchange supports both OAS 2.0 and OAS 3.0 specifications.
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -187,13 +191,22 @@ The name, icon, and description properties can only be changed in the Exchange a
-. (Optional) Enter the *API version*.
+. Click *Add version*. 
+. For API assets, follow the steps for the corresponding API asset type:
+* REST API - RAML: Provide a RAML API specification file. * RAML specifications must be a RAML file (.raml).
```
```suggestion
* REST API - RAML: Provide a RAML API specification file. A RAML specification must be a RAML file (.raml).
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -161,6 +164,7 @@ To create a custom asset:
 . Select the asset type *Custom* from the drop-down list.
 . (Optional) To share a file with users, choose the file.
 . (Optional) Select *Advanced* and edit the advanced settings: *GroupId*, *AssetId*, and *Version*. Exchange generates the group ID, asset ID, and version for you, and you can change these values here. The advanced settings are most often used to change the asset version.
+. (Optional) Select an option for *Lifecycle state*. The default is *Stable*.
```
```suggestion
. Optionally select an option for *Lifecycle state*. The default is *Stable*.
```

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -113,6 +115,7 @@ To create an API asset:
 If the file is a ZIP, the selected main file must be in the root directory of the ZIP file. If the file is not a ZIP or if the file is a ZIP file with only one main file, then the main file is selected automatically.
 +
 . (Optional) Select *Advanced* and edit the advanced settings: *GroupId*, *AssetId*, *Version*, and *API version*. Exchange generates the group ID, asset ID, and version for you, and you can change these values here. You can change an API asset's version (asset version) and API version separately. The advanced settings are most often used to change the asset version.
+. (Optional) Select an option for *Lifecycle state*. The default is *Stable*.
```
I guess you want to maintain consistency with the peer list items, but going forward let's convert these (Optional) steps to "If you want to..." statements or "Optionally, do this...."

kbostrom commented on modules/ROOT/pages/to-create-an-asset.adoc:
```diff
@@ -5,7 +5,7 @@ endif::[]
 :page-aliases: to-publish-an-asset.adoc
 
-To share an API, API Group, policy, example, template, or connector in either an Anypoint Exchange private instance or an Exchange public portal, create an asset of that asset type.
+To share an API, API Group, policy, example, template, or connector in either an Anypoint Exchange private instance or an Exchange public portal, create an asset of that type.
```
```suggestion
To share an API, API group, policy, example, template, or connector in either an Anypoint Exchange private instance or an Exchange public portal, create an asset of that type.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -106,20 +96,15 @@ However, these example values are not allowed in Exchange:
-that way. The table below outlines when to introduce a new version and when a 
-new version is not required. Typically when one clicks *Add new API operation* 
-a minor version increase occurs.
+It is not always possible to design APIs in that way. The table below outlines when to introduce a new version and when a new version is not required. Typically when one clicks *Add new API operation* a minor version increase occurs.
```
```suggestion
It is not always possible to design APIs in that way. The following table outlines when to introduce a new version and when a new version is not required. Typically, if you click *Add new API operation*, a minor version increase occurs.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -106,20 +96,15 @@ However, these example values are not allowed in Exchange:
-What is important is to manage the changes through a comprehensive versioning 
-strategy. This includes a multiple month deprecation schedule and documentation that 
-is up to date and describes the change in sufficient level of detail. 
+It is important is to manage the changes through a comprehensive versioning 
```
```suggestion
It is important is to manage the changes by using a comprehensive versioning 
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -106,20 +96,15 @@ However, these example values are not allowed in Exchange:
-strategy. This includes a multiple month deprecation schedule and documentation that 
-is up to date and describes the change in sufficient level of detail. 
+It is important is to manage the changes through a comprehensive versioning 
+strategy. This includes a multiple month deprecation schedule and documentation that is up to date and describes the change in sufficient level of detail. 
```
```suggestion
strategy. This includes a multiple-month deprecation schedule and documentation that is up to date and describes the change in a sufficient level of detail. 
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -106,20 +96,15 @@ However, these example values are not allowed in Exchange:
 
-An API should be designed for long term use, but because change is inevitable, the API 
-can never be completely stable. 
+An API should be designed for long term use, but because change is inevitable, the API can never be completely stable. 
```
```suggestion
An API should be designed for long-term use, but because change is inevitable, the API can never be completely stable. 
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -64,11 +57,8 @@ When you update the version element in the RAML specification and publish to Exc
-+
-. Change the API Version value. You can also change the Asset Version (the version of the Exchange asset).
+. Click *Publish to Exchange*.
+. Change the *API Version* value. You can also change the Asset Version (the version of the Exchange asset).
```
```suggestion
. Change the *API Version* value. You can also change *Asset Version* (the version of the Exchange asset).
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
-|===
-
-You can also change the version number for RAMLs as described in the next sections.
+Change the version number for RAMLs as described in the next sections.
```
Going forward, do not refer to other text units except by name, to ensure reusability.

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
+The Version and API version values must agree. You can only choose minor version updates for a given API version. To set a major Version number, also increase the API version. If you choose minor or patch version updates, the API versions should not changed.
 
-This opens the *Publish a new asset* window (as shown in <<deploy-a-new-api-asset>>).
+. (Optional) Select a *Lifecycle state*. Default is *Stable*.
```
```suggestion
. (Optional) Select a *Lifecycle state*. The default is *Stable*.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
+. (Required) In *Version* , enter a new version for the asset.
+. (Required) In *the *API version*, enter the version of the API specification.
+The Version and API version values must agree. You can only choose minor version updates for a given API version. To set a major Version number, also increase the API version. If you choose minor or patch version updates, the API versions should not changed.
```
```suggestion
The version and API version values must agree. You can choose only minor version updates for a given API version. To set a major version number, also increase the API version. If you choose minor or patch version updates, the API versions should not be changed.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
+. (Required) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
+. (Required) In *Version* , enter a new version for the asset.
+. (Required) In *the *API version*, enter the version of the API specification.
```
```suggestion
. (Required) In *API version*, enter the version of the API specification.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
+. (Required) Select a *Method* for the API specification.
+. (Required) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
+. (Required) In *Version* , enter a new version for the asset.
```
```suggestion
. (Required) In *Version*, enter a new version for the asset.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
+. From the asset details pane, click *Add version*. 
+. (Required) Select a *Method* for the API specification.
+. (Required) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
```
```suggestion
. Specify *Main file* if the uploaded file is a ZIP file. Otherwise, Exchange uses the same file name as the name of the file you upload. 
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
-image::ex-add-new-version.png[]
+. From the asset details pane, click *Add version*. 
+. (Required) Select a *Method* for the API specification.
+. (Required) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
```
```suggestion
. (Required) From *File upload*, click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. ZIP files are also accepted, with the file to upload residing in the ZIP file’s root directory.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
 
-image::ex-add-new-version.png[]
+. From the asset details pane, click *Add version*. 
+. (Required) Select a *Method* for the API specification.
```
Why are we specifying (Required) when typically all steps are required but those that we word using conditional wording (If you want to...) or "Optionally,..."?

kbostrom replied:
```suggestion
. (Required) Select a *Method* value for the API specification.
```

kbostrom commented on modules/ROOT/pages/to-change-raml-version.adoc:
```diff
@@ -24,36 +24,29 @@ NOTE: New asset versions count toward your organization's xref:to-create-an-asse
 == Deploy a New API Asset
 
-You can deploy an API asset directly into Exchange using the *New asset* button. In the *Publish a new asset* window, the Advanced section enables you to specify the version of the asset in the *Version* field, and the version of the API specification in the *API version* field.
+You can deploy an API asset directly into Exchange using the *Publish new asset* button. In the *Publish a new asset* window, the Advanced section enables you to specify the version of the asset in the *Version* field, and the version of the API specification in the *API version* field.
```
```suggestion
You can deploy an API asset directly to Exchange by using the *Publish new asset* button. In the *Publish a new asset* window, the *Advanced* section enables you to specify the version of the asset in the *Version* field, and the version of the API specification in the *API version* field.
```

kbostrom commented on modules/ROOT/pages/publish-an-http-api.adoc:
```diff
@@ -5,14 +5,16 @@ endif::[]
-. Give the API a name and in Asset Type, click *API - HTTP*.
+. In a Exchange, click *Publish new asset*.
+. Enter a name for the API
+. Choose an *HTTP API* from the *Asset types* drop-down menu.
```
```suggestion
. Choose an *HTTP API* value from the *Asset types* drop-down menu.
```

kbostrom commented on modules/ROOT/pages/publish-an-http-api.adoc:
```diff
@@ -5,14 +5,16 @@ endif::[]
+. Enter a name for the API
+. Choose an *HTTP API* from the *Asset types* drop-down menu.
+. Click *Publish*.
+. In the asset view, click the pencil icon to provide a description and click the check icon.
```
```suggestion
. In the asset view, click the pencil icon to provide a description, and then click the checkmark icon.
```
We should show rather than describe the "check" icon, or rename it "checkmark" to be just a little more clear.

kbostrom commented on modules/ROOT/pages/publish-an-http-api.adoc:
```diff
@@ -5,14 +5,16 @@ endif::[]
-. In a Private Exchange, click *New*.
-. Give the API a name and in Asset Type, click *API - HTTP*.
+. In a Exchange, click *Publish new asset*.
+. Enter a name for the API
```
```suggestion
. Enter a name for the API.
```

kbostrom commented on modules/ROOT/pages/publish-an-http-api.adoc:
```diff
@@ -5,14 +5,16 @@ endif::[]
 
-. In a Private Exchange, click *New*.
-. Give the API a name and in Asset Type, click *API - HTTP*.
+. In a Exchange, click *Publish new asset*.
```
```suggestion
. In Exchange, click *Publish new asset*.
```

kbostrom commented on modules/ROOT/pages/publish-an-http-api.adoc:
```diff
@@ -5,14 +5,16 @@ endif::[]
 :page-aliases: to-share-an-http-api.adoc
 
-The HTTP API lets Anypoint Exchange users with Private Exchange specify a placeholder for an endpoint. After you create the HTTP API endpoint, you can optionally manage the endpoint with API Manager.
+The HTTP API lets Anypoint Exchange users specify a placeholder for an endpoint. After you create the HTTP API endpoint, you can optionally manage the endpoint with API Manager.
```
```suggestion
The HTTP API enables Anypoint Exchange users to specify a placeholder for an endpoint. After you create the HTTP API endpoint, you can optionally manage the endpoint with API Manager.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+
+== Change Lifecycle State for a Version of an Asset
+
+The lifecycle state for a version of an asset can be changed depending on the following state promotions:
```
passive voice throughout; fix going forward:
```suggestion
You can change the lifecycle state of a version of an asset by using the following state promotions:
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+* `stable` to `deprecated`
+* `deprecated` to `stable`
+
+If an asset in the development state depends on one or more other assets in the development state, Exchange does not allow promoting that asset to the stable state. All its dependencies must first be promoted to stable.
```
```suggestion
If an asset in the development state depends on one or more other assets in the development state, Exchange does not promote that asset to the stable state. All of an asset's dependencies must first be promoted to a stable state before it can be promoted.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+. (Required) In *the *API version*, enter the version of the API specification.
+The Version and API version values must agree. You can only choose minor version updates for a given API version. To set a major Version number, also increase the API version. If you choose minor or patch version updates, the API versions should not changed.
+
+. (Optional) Select a *Lifecycle state*. The default is *Stable*.
```
```suggestion
. (Optional) Select a *Lifecycle state* value. The default is *Stable*.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
+. (Required) In *Version* , enter a new version for the asset.
+. (Required) In *the *API version*, enter the version of the API specification.
+The Version and API version values must agree. You can only choose minor version updates for a given API version. To set a major Version number, also increase the API version. If you choose minor or patch version updates, the API versions should not changed.
```
```suggestion
The version and API version values must agree. You can choose only minor version updates for a given API version. To set a major version number, also increase the API version. If you choose minor or patch version updates, the API versions should not be changed.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+. (Required for APIs) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
+. (Required) In *Version* , enter a new version for the asset.
+. (Required) In *the *API version*, enter the version of the API specification.
```
```suggestion
. (Required) In *API version*, enter the version of the API specification.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+. (Required for APIs) Select a *Method* for the API specification.
+. (Required for APIs) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
+. (Required) In *Version* , enter a new version for the asset.
```
```suggestion
. (Required) In *Version*, enter a new version for the asset.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+. Click *Add new version*.
+. (Required for APIs) Select a *Method* for the API specification.
+. (Required for APIs) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
+. Specify the *Main file* if the uploaded file is a zip. Otherwise, Exchange uses the same file name as the file you upload. 
```
```suggestion
. Specify a *Main file* value if the uploaded file is a ZIP. Otherwise, Exchange uses the same file name as the name of the file you upload. 
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+. Click *Manage versions* in the expanded asset details pane.
+. Click *Add new version*.
+. (Required for APIs) Select a *Method* for the API specification.
+. (Required for APIs) From *File upload* click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. Zip files are also accepted with the file to upload residing in the zip file’s root directory.
```
```suggestion
. (Required for APIs) From *File upload*, click *Choose file*. For OAS, choose an OAS file with the .yaml or .json file types. For RAML, choose a RAML file with the .raml file type. ZIP files are also accepted. with the file to upload residing in the ZIP file’s root directory.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+
+. Click *Manage versions* in the expanded asset details pane.
+. Click *Add new version*.
+. (Required for APIs) Select a *Method* for the API specification.
```
```suggestion
. (Required for APIs) Select a *Method* value for the API specification.
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+. Click *Manage versions* in the expanded asset details pane.
+. Click the three dots menu and choose *Version detail* from the menu.
+-- 
+== Add New Version of an Asset
```
```suggestion
== Add a New Version of an Asset
```

kbostrom commented on modules/ROOT/pages/manage-versions.adoc:
```diff
@@ -0,0 +1,62 @@
+== View Version Detail
+You can view the following information about a version of an asset:
+
+- Name of the owner who published the asset
+- Date the version was published
+- Lifecycle status
```
Use proper bullet-list formatting: *

kbostrom reviewed (approved):
Completed previously started edit review.
Did not reverify already-edited texts due to time constraints.
